### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by Guillemdb**
Refactor ml-mining into a separate library.

### CODE REVIEW

**Comment by Guillemdb** [comment]
When I figure out how  to make the SKIP_SPARK_TESTS=1  in the .travis.yaml file work, I will make the commits look sexy.

File: .travis.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ _install: &_install
   - sudo apt install -y libsnappy-dev
   - travis_retry make bblfsh-start
   - pip install --upgrade pip cython codecov
-  - ML_MINING_SETUP_INCLUDE_TESTS=1  pip install ".[tf, pandas]"
+  - ML_MINING_SETUP_INCLUDE_TESTS=1 SKIP_SPARK_TESTS=1 pip install ".[tf, pandas]"
```
Comment (vmarkovtsev) [inline .travis.yml]: Nope, the should be skipped unconditionally.

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (Guillemdb) [reply .travis.yml]: Sure, I will default it to True, and leave the flag in case someone feels like debugging that hell.

Comment (vmarkovtsev) [reply .travis.yml]: ok

**Comment by vmarkovtsev** [comment]
@Guillemdb ping

**Comment by Guillemdb** [comment]
I got caught with the pyspark thing. I will upload the fix and rebase the commits before tomorrow morning.

**Comment by zurk** [comment]
wow, commit `Fix CI configuration` in the middle have a weird description: https://github.com/src-d/ml-mining/pull/3/commits/37834da244b8ed057118380d8a5b315a32bbb793
Please, fix :)

EDIT: and the first one too actually

File: .travis.yml [hunk 2]
```diff
@@ -14,12 +14,12 @@ stages:
...
 _install: &_install
   - sudo apt update -y
-  - sudo apt install -y libsnappy-dev
+  - sudo apt install -y libsnappy-dev gzip
   - travis_retry make bblfsh-start
   - pip install --upgrade pip cython codecov
-  - ML_MINING_SETUP_INCLUDE_TESTS=1  pip install ".[tf, pandas]"
+  - ML_MINING_SETUP_INCLUDE_TESTS=1 pip install ".[tf, pandas]"
   - cd $(pip show sourced.ml.mining|grep Location|cut -d' ' -f2)/sourced/ml/mining
-  - find . -wholename "*/tests/*" -type d -exec chmod 555 {} \;
```
Comment (zurk) [inline .travis.yml]: you should keep this line.
We should not tough test directory while testing.

**Review by zurk — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 2 above)
Comment (Guillemdb) [reply .travis.yml]: but the tests uncompress tar files. If I freeze this I get permission errors. I also tried to change the permissions of the specific files to be decompressed, but I couldn't make it right.

Comment (zurk) [reply .travis.yml]: but you can read it and decompress to a temporary location.

Comment (Guillemdb) [reply .travis.yml]: I tried, but I don't know how to make the decompression process, which is called using `subprocess.check_call(["gzip", "-fdk", path + ".gz"])` to write to a temporary file.

Do you know how to redirect the target of a subprocess call? otherwis I will need to rewrite all the tests, and I am not sure it is worth it...

Comment (Guillemdb) [reply .travis.yml]: Solved

**Review by zurk — APPROVED** [review]

**Comment by vmarkovtsev** [comment]
Closing because there is no ability to review 107 files and no resources.

**Pull request closed without merging**